includes/class-ast-staging.php

```diff
@@ -0,0 +1,215 @@
+		if (
+			self::is_duplicate_site() &&
+			current_user_can( 'manage_options' ) &&
+			self::get_autoshare_site_url_lock_key() !== get_option( 'autoshare_ignore_duplicate_siteurl_notice' )
+		) {
+			$ignore_url = wp_nonce_url( add_query_arg( 'autoshare_duplicate_site', 'ignore' ), 'ast_duplicate_site', '_astnonce' );
+			$update_url = wp_nonce_url( add_query_arg( 'autoshare_duplicate_site', 'update' ), 'ast_duplicate_site', '_astnonce' );
+			?>
+			<div class="notice notice-error">
+				<p>
+					<?php
+					printf(
+						// translators: 1$-2$: opening and closing <strong> tags. 3$-4$: Opening and closing link to production URL. 5$: Production URL.
+						esc_html__( 'It looks like this site has moved or is a duplicate site. %1$sAutoshare for Twitter%2$s has disabled publish tweets on this site to prevent tweets from a staging or test environment. %1$sAutoshare for Twitter%2$s considers %3$s%5$s%4$s to be the site\'s URL. ', 'autoshare-for-twitter' ),
+						'<strong>',
+						'</strong>',
+						'<a href="' . esc_url( self::get_site_url_from_source( 'autoshare' ) ) . '" target="_blank">',
+						'</a>',
+						esc_url( self::get_site_url_from_source( 'autoshare' ) )
+					)
+					?>
+				</p>
+				<p>
+					<a class="button button-primary" href="<?php echo esc_url( $ignore_url ); ?>">
+						<?php esc_html_e( 'Dismiss this (but don\'t enable publish tweets)', 'autoshare-for-twitter' ); ?>
+					</a>
+					<a class="button" href="<?php echo esc_url( $update_url ); ?>">
+						<?php esc_html_e( 'Enable publish tweets', 'autoshare-for-twitter' ); ?>
+					</a>
+				</p>
+			</div>
+			<?php
+		}
+	}
+
+	/**
+	 * Determines if this is a testing/staging site.
+	 *
+	 * Checks if the WordPress site URL is the same as the URL Autoshare considers the live URL.
+	 *
+	 * @since 1.2.0
+	 * @return bool Whether the site is a staging URL or not.
+	 */
+	public static function is_duplicate_site() {
+		$wp_site_url_parts  = wp_parse_url( self::get_site_url_from_source( 'current_wp_site' ) );
+		$ast_site_url_parts = wp_parse_url( self::get_site_url_from_source( 'autoshare' ) );
+
+		if ( ! isset( $wp_site_url_parts['path'] ) && ! isset( $ast_site_url_parts['path'] ) ) {
+			$paths_match = true;
+		} elseif ( isset( $wp_site_url_parts['path'] ) && isset( $ast_site_url_parts['path'] ) && $wp_site_url_parts['path'] === $ast_site_url_parts['path'] ) {
+			$paths_match = true;
+		} else {
+			$paths_match = false;
+		}
+
+		if ( isset( $wp_site_url_parts['host'] ) && isset( $ast_site_url_parts['host'] ) && $wp_site_url_parts['host'] === $ast_site_url_parts['host'] ) {
+			$hosts_match = true;
+		} else {
+			$hosts_match = false;
+		}
+
+		// Check the host and path, do not check the protocol/scheme to avoid issues.
+		if ( $paths_match && $hosts_match ) {
+			$is_duplicate = false;
+		} else {
+			$is_duplicate = true;
+		}
+
+		/**
+		 * Filters value of "Is staging site?".
+		 *
+		 * @since 1.2.0
+		 * @param boolean $is_duplicate Is staging site?.
```

Comment:
```suggestion
		 * @param boolean $is_duplicate Whether the site is a staging URL or not.
```

As there are staging, development and local sites, instead of `$is_duplicate` it might be clearer to use `$is_production` and reverse the logic of the checks.